src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -441,6 +441,79 @@ public function get($id, $invalidBehavior = ContainerInterface::EXCEPTION_ON_INV
     }
 
     /**
+     * {@inheritdoc}
+     *
+     * @param bool $reload whether or not reload the type map
+     */
+    public function getServiceIdsOfType($class, $reload = true)
+    {
+        if ($reload) {
+            $this->populateAvailableTypes();
+        }
+
+        return parent::getServiceIdsOfType($class);
+    }
+
+    /**
+     * Populates the list of available types.
+     *
+     * @return array the type map
+     */
+    public function populateAvailableTypes()
```

Comment:
Could we make `$this->typeMap` null by default (it actually is already) and then in `getServiceIdsOfType`, call this *private* function if `null === $this->typeMap`. Basically, *this* class should be able to manage the fact that all the types should be populated once, at the beginning of `getServiceIdsOfType`.